apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java

```diff
@@ -23,39 +27,60 @@
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
     final TextView txtResponse = (TextView) findViewById(R.id.txt_response);
-    SampleApplication application = (SampleApplication) getApplication();
-    application.githuntApiService()
-        .githuntFeed(new OperationRequest<>(
-            new FeedQuery(FeedQuery.Variables.builder()
-                .limit(10)
-                .type(FeedType.HOT)
-                .build())))
+    final RecyclerView responses = (RecyclerView) findViewById(R.id.responses);
+
+    final SampleApplication application = (SampleApplication) getApplication();
+
+    final Subject<Integer> upvoteObserver = PublishSubject.create();
+
+    final PostsAdapter adapter = new PostsAdapter(upvoteObserver);
+    responses.setAdapter(adapter);
+    responses.setLayoutManager(new LinearLayoutManager(this));
+
+    upvoteObserver.subscribe(new SimpleObserver<Integer>() {
+      @Override public void onNext(Integer postId) {
+        final Upvote.Variables variables = Upvote.Variables.builder()
+            .postId(postId)
+            .build();
+
+        application.postsService()
+            .upvote(new Upvote(variables))
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(new SimpleObserver<Response<Upvote.Data>>() {
+              @Override public void onNext(Response<Upvote.Data> response) {
+                adapter.upvotePost(response.data().upvotePost());
+              }
+            });
+      }
+    });
+
+    application.postsService()
+        .allPosts(new AllPosts())
         .subscribeOn(Schedulers.io())
         .observeOn(AndroidSchedulers.mainThread())
-        .subscribe(new Observer<Response<FeedQuery.Data>>() {
-          @Override public void onSubscribe(Disposable d) {
-          }
+        .subscribe(new SimpleObserver<Response<AllPosts.Data>>() {
+          @Override public void onNext(Response<AllPosts.Data> response) {
+            txtResponse.setVisibility(View.GONE);
```

Comment:
Be aware of potentially memory leaks here, as you have implicit ref to main activity, and pass this ref to background thread as observer, so main activity will be kept as long as this subscription is alive